platforms/chibios/drivers/analog.c

```diff
@@ -240,6 +261,103 @@ __attribute__((weak)) adc_mux pinToMux(pin_t pin) {
+#elif defined(STM32L4XX)
+        case A0: return TO_MUX( ADC_CHANNEL_IN5,  0 ); // Can also be ADC2 in some cases
+        case A1: return TO_MUX( ADC_CHANNEL_IN6,  0 ); // Can also be ADC2 in some cases
+        case A2: return TO_MUX( ADC_CHANNEL_IN7,  0 ); // Can also be ADC2
+        case A3: return TO_MUX( ADC_CHANNEL_IN8,  0 ); // Can also be ADC2
+        case A4: return TO_MUX( ADC_CHANNEL_IN9,  0 ); // Can also be ADC2
+        case A5: return TO_MUX( ADC_CHANNEL_IN10, 0 ); // Can also be ADC2
+        case A6: return TO_MUX( ADC_CHANNEL_IN11, 0 ); // Can also be ADC2
+        case A7: return TO_MUX( ADC_CHANNEL_IN12, 0 ); // Can also be ADC2
+        case B0: return TO_MUX( ADC_CHANNEL_IN15, 0 ); // Can also be ADC2
+        case B1: return TO_MUX( ADC_CHANNEL_IN16, 0 ); // Can also be ADC2
+        case C0: return TO_MUX( ADC_CHANNEL_IN1,  0 ); // Can also be ADC2 or ADC3
+        case C1: return TO_MUX( ADC_CHANNEL_IN2,  0 ); // Can also be ADC2 or ADC3
+        case C2: return TO_MUX( ADC_CHANNEL_IN3,  0 ); // Can also be ADC2 or ADC3
+        case C3: return TO_MUX( ADC_CHANNEL_IN4,  0 ); // Can also be ADC2 or ADC3
+        case C4: return TO_MUX( ADC_CHANNEL_IN13, 0 ); // Can also be ADC2
+        case C5: return TO_MUX( ADC_CHANNEL_IN14, 0 ); // Can also be ADC2
+#    if STM32_ADC_USE_ADC3
+        case F3:  return TO_MUX( ADC_CHANNEL_IN6,  2 );
+        case F4:  return TO_MUX( ADC_CHANNEL_IN7,  2 );
+        case F5:  return TO_MUX( ADC_CHANNEL_IN8,  2 );
+        case F6:  return TO_MUX( ADC_CHANNEL_IN9,  2 );
+        case F7:  return TO_MUX( ADC_CHANNEL_IN10, 2 );
+        case F8:  return TO_MUX( ADC_CHANNEL_IN11, 2 );
+        case F9:  return TO_MUX( ADC_CHANNEL_IN12, 2 );
+        case F10: return TO_MUX( ADC_CHANNEL_IN13, 2 );
+#    endif
+#elif defined(STM32G4XX)
+#    if STM32_ADC_USE_ADC1
+        case A0:  return TO_MUX( ADC_CHANNEL_IN1,  0 );
+        case A1:  return TO_MUX( ADC_CHANNEL_IN2,  0 );
+        case A2:  return TO_MUX( ADC_CHANNEL_IN3,  0 );
+        case A3:  return TO_MUX( ADC_CHANNEL_IN4,  0 );
+        case B0:  return TO_MUX( ADC_CHANNEL_IN15, 0 );
+        case B1:  return TO_MUX( ADC_CHANNEL_IN12, 0 );
+        case B11: return TO_MUX( ADC_CHANNEL_IN14, 0 );
+        case B12: return TO_MUX( ADC_CHANNEL_IN11, 0 );
+        case B14: return TO_MUX( ADC_CHANNEL_IN5,  0 );
+        case C0:  return TO_MUX( ADC_CHANNEL_IN6,  0 );
+        case C1:  return TO_MUX( ADC_CHANNEL_IN7,  0 );
+        case C2:  return TO_MUX( ADC_CHANNEL_IN8,  0 );
+        case C3:  return TO_MUX( ADC_CHANNEL_IN9,  0 );
+        case F0:  return TO_MUX( ADC_CHANNEL_IN10, 0 );
+#    endif
+#    if STM32_ADC_USE_ADC2
+        case A0:  return TO_MUX( ADC_CHANNEL_IN1,  1 );
+        case A1:  return TO_MUX( ADC_CHANNEL_IN2,  1 );
+        case A4:  return TO_MUX( ADC_CHANNEL_IN17, 1 );
+        case A5:  return TO_MUX( ADC_CHANNEL_IN13, 1 );
```

Comment:
Looks like this won't compile if you enable both `STM32_ADC_USE_ADC1` and `STM32_ADC_USE_ADC2` (`case A0:` would be duplicated), and this configuration would be needed if you want to cover all analog pins on even something smaller like G431.

If we don't want to support the configuration where `STM32_ADC_USE_ADC2` is enabled, but `STM32_ADC_USE_ADC1` is not, we could just drop all these `#if`s and specify the ADC with the lowest index for each supported pin.  The code for most other chips is already like that (those `// Can also be ADC2` comments refer to exactly that situation).

Apparently all G4xx chips currently supported by ChibiOS have the same set of `STM32_HAS_GPIO[A-G]` enabled, so there should not be a problem with something like `GPIOF` being undefined, and referencing some ADC index which is not actually enabled should be safe (`adc_read()` would return 0 due to `intToADCDriver()` returning NULL in that case).

Reply:
Yeah, I thought about that right when the compile failed, I'll modify the logic to enable the pins on the ADC with the lowest index and add that comment. For some reason, I skipped that and my brain thought about the channels instead of the shared pins..... ma bad.
The goal here was to enable the user to enable any `ADCx` (even multiple ones) and maximize the number of available ADC pins. Indeed this won't work, will apply changes to fix this.